commercelayer/resource_braintree_gateway.go

```diff
@@ -0,0 +1,182 @@
+package commercelayer
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	commercelayer "github.com/incentro-dc/go-commercelayer-sdk/api"
+)
+
+func resourceBraintreeGateway() *schema.Resource {
+	return &schema.Resource{
+		Description: "Configuring a Braintree payment gateway for a market lets you safely process payments " +
+			"through Braintree. The Braintree gateway is compliant with the PSD2 European regulation " +
+			"so that you can implement a payment flow that supports SCA and 3DS2 by using the " +
+			"Braintree official JS SDK and libraries.",
+		ReadContext:   resourceBraintreeGatewayReadFunc,
+		CreateContext: resourceBraintreeGatewayCreateFunc,
+		UpdateContext: resourceBraintreeGatewayUpdateFunc,
+		DeleteContext: resourceBraintreeGatewayDeleteFunc,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The braintree payment unique identifier",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"type": {
+				Description: "The resource type",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"attributes": {
+				Description: "Resource attributes",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Description: "The payment gateway's internal name.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"merchant_account_id": {
+							Description: "The gateway merchant account ID.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"merchant_id": {
+							Description: "The gateway merchant ID.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"public_key": {
+							Description: "The gateway API public key.",
+							Type:        schema.TypeString,
+							Required:    true,
+						},
+						"private_key": {
```

Comment:
arent you missing the `descriptor_name`, `descriptor_phone` and `descriptor_url` fields?